src/garage/torch/algos/maml/maml.py

```diff
@@ -0,0 +1,478 @@
+"""Model-Agnostic Meta-Learning (MAML) algorithm implementation for RL."""
+import collections
+
+from dowel import tabular
+import numpy as np
+import torch
+
+from garage.misc import tensor_utils
+from garage.sampler import OnPolicyVectorizedSampler
+from garage.tf.samplers import BatchSampler
+from garage.torch.algos import _Default, make_optimizer
+from garage.torch.optimizers import ConjugateGradientOptimizer, DiffSGD
+from garage.torch.utils import update_module_params
+
+
+class MAML:
+    """Model-Agnostic Meta-Learning (MAML).
+
+    Args:
+        env (garage.envs.GarageEnv): A gym environment.
+        policy (garage.torch.policies.Policy): Policy.
+        baseline (garage.np.baselines.Baseline): The baseline.
+        meta_optimizer (Union[torch.optim.Optimizer, tuple]):
+            Type of optimizer.
+            This can be an optimizer type such as `torch.optim.Adam` or a tuple
+            of type and dictionary, where dictionary contains arguments to
+            initialize the optimizer e.g. `(torch.optim.Adam, {'lr' = 1e-3})`.
+        meta_batch_size (int): Number of tasks sampled per batch.
+        inner_lr (double): Adaptation learning rate.
+        num_grad_updates (int): Number of adaptation gradient steps.
+        inner_algo (garage.torch.algos.VPG): The inner algorithm used for
+            computing loss.
+
+    """
+
+    def __init__(self,
+                 inner_algo,
+                 env,
+                 policy,
+                 baseline,
+                 meta_optimizer,
+                 meta_batch_size=40,
+                 inner_lr=0.1,
+                 num_grad_updates=1):
+        if policy.vectorized:
+            self.sampler_cls = OnPolicyVectorizedSampler
+        else:
+            self.sampler_cls = BatchSampler
+
+        self.max_path_length = inner_algo.max_path_length
+        self._policy = policy
+        self._env = env
+        self._baseline = baseline
+        self._num_grad_updates = num_grad_updates
+        self._meta_batch_size = meta_batch_size
+        self._inner_algo = inner_algo
+        self._inner_optimizer = DiffSGD(self._policy, lr=inner_lr)
+        self._meta_optimizer = make_optimizer(meta_optimizer,
+                                              policy,
+                                              lr=_Default(1e-3),
+                                              eps=_Default(1e-5))
+        self._episode_reward_mean = [
```

Comment:
why is  this needed? AFAIK it's only in a couple off-policy algorithms to compare to openai/baselines implementations

Reply:
I removed it.